### PR TITLE
Devsecops

### PR DESCRIPTION
**Pull request by ayman-m**
## Status
- [x] In Progress
- [ ] Ready
- [ ] In Hold - (Reason for hold)

## Related Issues
fixes: link to the issue

## Description
A pack of DevSecOps integrations and playbooks, more information in pack's README file

## Screenshots
Screenshots in pack's README file

## Minimum version of Demisto
- [ ] 5.0.0
- [x] 5.5.0
- [ ] 6.0.0

## Does it break backward compatibility?
   - [ ] Yes
       - Further details:
   - [x] No

## Must have
- [ ] Tests
- [x] Documentation

### CODE REVIEW

**Review by yaakovi — CHANGES_REQUESTED** [review]
following our session,
please remove the Github-related changes to a separate PR and apply them to the current Github integration.

when opening the new PR, link it to this one.
after you are done, we will take it from there.

thank you.

**Comment by ayman-m** [comment]
Hello @yaakovi, I have pushed a new commit after reviewing your comments

**Comment by ayman-m** [comment]
The Github integration update [PR](https://github.com/demisto/content/pull/9972)

**Comment by ayman-m** [comment]
Hello @yaakovi, I have pushed a new Integration to this Pack,

**Comment by ChanochShayner** [comment]
Following our conversation from Thursday, here are comments we talked about:
 - Please run the demisto-sdk format command on the files you have in PR.
 - You are missing 3 incident types (which appear within the incident fields) and one type indicator.
 - The indicator - commit_id associated with incident types instead of indicator types.
The names of all incident/indicator/fields/types should start with the name of the pack - for example, `commitid` -> `devsecops_commitid`
Please correct these comments and update me so I can go through all the files in PR.
Thanks!!

**Comment by ayman-m** [comment]
>     * `devsecops_commitid`

@ChanochShayner , I've pushed two new commits that reflect the updates based on the findings above

File: Packs/DevSecOps/IncidentFields/DevSecOps_After_Commit_ID.json [hunk 1]
```diff
@@ -0,0 +1,30 @@
+{
+    "id": "incident_aftercommitid",
```
Comment (ChanochShayner) [inline Packs/DevSecOps/IncidentFields/DevSecOps_After_Commit_ID.json:2]:
```suggestion
    "id": "incident_devsecopsaftercommitid",
``` 
Same in all incident fields/types indicator fields/types

File: Packs/DevSecOps/IncidentFields/DevSecOps_After_Commit_ID.json [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+    "version": -1,
+    "modified": "2020-08-09T11:01:34.448790364+04:00",
+    "name": "After Commit ID",
```
Comment (ChanochShayner) [inline Packs/DevSecOps/IncidentFields/DevSecOps_After_Commit_ID.json:5]:
```suggestion
    "name": "Dev Sec Ops After Commit ID",
```
Same in all incident fields/types indicator fields/types

File: Packs/DevSecOps/IncidentFields/DevSecOps_After_Commit_ID.json [hunk 3]
```diff
@@ -0,0 +1,30 @@
...
+    "ownerOnly": false,
+    "cliName": "aftercommitid",
```
Comment (ChanochShayner) [inline Packs/DevSecOps/IncidentFields/DevSecOps_After_Commit_ID.json:7]:
```suggestion
    "cliName": "devsecopsaftercommitid",
```
Same in all incident fields/types indicator fields/types

File: Packs/DevSecOps/IncidentFields/DevSecOps_After_Commit_ID.json [hunk 4]
```diff
@@ -0,0 +1,30 @@
...
+    "neverSetAsRequired": false,
+    "isReadOnly": false,
+    "useAsKpi": false,
+    "locked": false,
+    "system": false,
+    "content": true,
+    "group": 0,
+    "hidden": false,
+    "associatedTypes": [
+        "New Git PR"
```
Comment (ChanochShayner) [inline Packs/DevSecOps/IncidentFields/DevSecOps_After_Commit_ID.json:21]:
```suggestion
        "Dev Sec Ops New Git PR"
```
Same in all incident fields/types indicator fields/types

File: Packs/DevSecOps/Integrations/DockerEngine/DockerEngine.yml [hunk 5]
```diff
@@ -0,0 +1,1997 @@
...
+configuration:
+- display: Server URL (e.g. https://www.example.com:1000)
+  name: url
+  required: true
+  type: 0
+- display: Docker Client Certificate
+  hidden: false
+  name: client_certificate
+  required: true
+  type: 12
```
Comment (ChanochShayner) [inline Packs/DevSecOps/Integrations/DockerEngine/DockerEngine.yml:14]:
```suggestion
  type: 4
```
Please use encrypted type - 4

File: Packs/DevSecOps/Integrations/DockerEngine/DockerEngine.yml [hunk 6]
```diff
@@ -0,0 +1,1997 @@
...
+- display: Docker Client Private Key
+  hidden: false
+  name: client_key
+  required: true
+  type: 12
```
Comment (ChanochShayner) [inline Packs/DevSecOps/Integrations/DockerEngine/DockerEngine.yml:19]:
```suggestion
  type: 4
```
Same here

File: Packs/DevSecOps/Integrations/GitLab/GitLab.yml [hunk 7]
```diff
@@ -0,0 +1,9 @@
+commonfields:
+  id: cc4e029c-d66e-4bfc-88ef-ce398f3a88a9
+  version: -1
+enabled: false
+script: '-'
+subtype: python3
+system: false
+timeout: '0'
+type: python
```
Comment (ChanochShayner) [inline Packs/DevSecOps/Integrations/GitLab/GitLab.yml:9]: This file does not look good, please fix it.

File: Packs/DevSecOps/Playbooks/DevSecOps_-_Fetch_PR_-_Triage.yml [hunk 8]
```diff
@@ -0,0 +1,2402 @@
...
+    simple: "false"
+  required: false
+  description: ""
+  playbookInputQuery:
+outputs: []
+tests:
+- GitLab-test-playbook
+- LGTM-test-playbook
+- playbook-MinIO-Test
+fromversion: 1.0.0
```
Comment (ChanochShayner) [inline Packs/DevSecOps/Playbooks/DevSecOps_-_Fetch_PR_-_Triage.yml:2402]:
```suggestion
fromversion: 5.0.0
```
The version should be consistent across all files in the Pack
(It's currently 5.0.0) Please fix this.

File: Packs/DevSecOps/Playbooks/DevSecOps_-_LGTM_-_Analysis_-_SubPB.yml [hunk 9]
```diff
@@ -0,0 +1,1851 @@
...
+  type: unknown
+- contextPath: LGTM.code_review_result.languages.alerts
+  type: unknown
+- contextPath: LGTM.code_review_result.languages.alerts.query.properties
+  type: unknown
+tests:
+- GitLab-test-playbook
+- LGTM-test-playbook
+- playbook-MinIO-Test
+fromversion: 1.0.0
```
Comment (ChanochShayner) [inline Packs/DevSecOps/Playbooks/DevSecOps_-_LGTM_-_Analysis_-_SubPB.yml:1851]:
```suggestion
fromversion: 5.0.0
```
The version should be consistent across all files in the Pack
(It's currently 5.0.0) Please fix this.

File: Packs/DevSecOps/Playbooks/DevSecOps_-_Prisma_-_Analysis_-SubPB.yml [hunk 10]
```diff
@@ -0,0 +1,320 @@
...
+  value: {}
+  required: false
+  description: Prisma Cloud Check Run Title
+  playbookInputQuery:
+outputs: []
+tests:
+- GitLab-test-playbook
+- LGTM-test-playbook
+- playbook-MinIO-Test
+fromversion: 1.0.0
```
Comment (ChanochShayner) [inline Packs/DevSecOps/Playbooks/DevSecOps_-_Prisma_-_Analysis_-SubPB.yml:320]:
```suggestion
fromversion: 5.0.0
```
The version should be consistent across all files in the Pack
(It's currently 5.0.0) Please fix this.

File: Packs/DevSecOps/TestPlaybooks/GitLab-test-playbook.yml [hunk 11]
```diff
@@ -0,0 +1,96 @@
+id: GitLab-test-playbook
+name: GitLab-test-playbook
+version: -1
+fromversion: 4.5.0
```
Comment (ChanochShayner) [inline Packs/DevSecOps/TestPlaybooks/GitLab-test-playbook.yml:4]:
```suggestion
fromversion: 5.0.0
```
The version should be consistent across all files in the Pack
(It's currently 5.0.0) Please fix this.

File: Packs/DevSecOps/TestPlaybooks/LGTM-test-playbook.yml [hunk 12]
```diff
@@ -0,0 +1,1122 @@
+id: LGTM-test-playbook
+name: LGTM-test-playbook
+version: -1
+fromversion: 4.5.0
```
Comment (ChanochShayner) [inline Packs/DevSecOps/TestPlaybooks/LGTM-test-playbook.yml:4]:
```suggestion
fromversion: 5.0.0
```
The version should be consistent across all files in the Pack
(It's currently 5.0.0) Please fix this.

File: Packs/DevSecOps/TestPlaybooks/MinIO-test-playbook.yml [hunk 13]
```diff
@@ -0,0 +1,440 @@
+id: playbook-MinIO-Test
+name: playbook-MinIO-Test
+version: -1
+fromversion: 4.5.0
```
Comment (ChanochShayner) [inline Packs/DevSecOps/TestPlaybooks/MinIO-test-playbook.yml:4]:
```suggestion
fromversion: 5.0.0
```
The version should be consistent across all files in the Pack
(It's currently 5.0.0) Please fix this.

File: Packs/DevSecOps/pack_metadata.json [hunk 14]
```diff
@@ -0,0 +1,27 @@
+{
+    "name": "DevSecOps",
+    "description": "DevSecOps CI/CD Orchestration Integration Pack.",
+    "support": "community",
+    "currentVersion": "1.0.0",
+    "author": "Ayman Mahmoud",
+    "url": "https://www.paloaltonetworks.com/cortex",
```
Comment (ChanochShayner) [inline Packs/DevSecOps/pack_metadata.json:7]: The URL here should be the product in the pack, please remove it.

File: Tests/conf.json [hunk 15]
```diff
@@ -3033,6 +3033,15 @@
...
             "fromversion": "5.5.0"
+        },
+        {
+            "playbookID": "GitLab-test-playbook"
+        },
+        {
+            "playbookID": "LGTM-test-playbook"
+        },
+        {
+            "playbookID": "playbook-MinIO-Test"
```
Comment (ChanochShayner) [inline Tests/conf.json:3044]: Please remove the changes from this file.

File: Packs/DevSecOps/README.md [hunk 16]
```diff
@@ -0,0 +1,68 @@
...
+![](doc_files/PR_-_Triage_PB.png)
+
+* A Github App Task analysis Playbook that parses the received Github App Task delivery, create vulnerability indicators if any vuln found and then link the App Task incident&indicators to the PR incident
+![](doc_files/App%20Task%20Analysis%20PB.png)
+
+* The App Task analysis Playbook has two subplaybooks:
+    * LGTM Task Analysis Playbook, the parses the LGTM task deliveries and create SAST vuln indicators if any
+![](doc_files/LGTM%20Analysis%20PB.png) 
+    * Prisma Cloud Task Analysis Playbook, the parses the Prisma task deliveries and create IAST vuln indicators if any
+![](doc_files/Prisma%20Analysis%20PB.png)
```
Comment (ChanochShayner) [inline Packs/DevSecOps/README.md:68]: It seems that the file has not been updated for recent changes, for example, there is only one incident type.
Please update it.

File: Packs/DevSecOps/Integrations/MinIO/README.md [hunk 17]
```diff
@@ -0,0 +1,475 @@
+An Integration with MinIO Object Storage
+This integration was integrated and tested with version xx of MinIO.
```
Comment (ChanochShayner) [inline Packs/DevSecOps/Integrations/MinIO/README.md:2]:
```suggestion
This integration was integrated and tested with version ? of MinIO.
```
Please fill in the relevant version (if there is one)

File: Packs/DevSecOps/Integrations/GitLab/README.md [hunk 18]
```diff
@@ -0,0 +1,902 @@
...
+```!gitlab-projects-request-access id=21898188 using=Asteroid```
+
+#### Context Example
+```json
+{
+    "GitLab": {
+        "AccessRequests": {
+            "avatar_url": "https://secure.gravatar.com/avatar/bcecfc2b23ff4a3962520685ccf046cc?s=80&d=identicon",
+            "id": 7475865,
+            "name": "Omar Elkhayam",
```
Comment (ChanochShayner) [inline Packs/DevSecOps/Integrations/GitLab/README.md:436]:
```suggestion
            "name": "?",
```
We do not want specific people's names to be in our files.

File: Packs/DevSecOps/Integrations/GitLab/README.md [hunk 19]
```diff
@@ -0,0 +1,902 @@
...
+            "requested_at": "2020-10-25T12:50:05.865Z",
+            "state": "active",
+            "username": "OmarElkhayam",
```
Comment (ChanochShayner) [inline Packs/DevSecOps/Integrations/GitLab/README.md:439]: same here

**Review by ChanochShayner — COMMENTED** [review]
Looks great!!
1 - Are the versions of the files in the Pack intentionally version 5 rather than version 6?
2 - You have a lot of files with run examples (mostly readme files) that have really specific details about people or codes.
Please remove them, you can use names that are understood to be sample/test only.

File: Tests/conf.json [hunk 20]
```diff
@@ -3409,4 +3403,4 @@
             }
         }
     }
-}
+}
```
Comment (ChanochShayner) [inline Tests/conf.json:3406]: Could you please remove the changes (even if they are just blank lines) from this file - conf.json?
It causes conflicts.

File: Packs/DevSecOps/TestPlaybooks/LGTM-test-playbook.yml (hunk 12 above)
Comment (ChanochShayner) [inline Packs/DevSecOps/TestPlaybooks/LGTM-test-playbook.yml:4]:
```suggestion
fromversion: 6.0.0
```

File: Packs/DevSecOps/TestPlaybooks/MinIO-test-playbook.yml (hunk 13 above)
Comment (ChanochShayner) [inline Packs/DevSecOps/TestPlaybooks/MinIO-test-playbook.yml:4]:
```suggestion
fromversion: 6.0.0
```

File: Packs/DevSecOps/TestPlaybooks/GitLab-test-playbook.yml (hunk 11 above)
Comment (ChanochShayner) [inline Packs/DevSecOps/TestPlaybooks/GitLab-test-playbook.yml:4]:
```suggestion
fromversion: 6.0.0
```

**Review by ChanochShayner — COMMENTED** [review]
Please go over the validation and lint errors and fix them.
In layouts, you changed the versions from - 
"fromVersion": "5.0.0",
"toVersion": "5.9.9",
To -
"fromVersion": "6.0.0",

In layouts there is a difference between the versions, so to change a version you need to remove the change and run the command - `demisto-sdk convert-layout -i Packs / TestPack1 -fts`

**Comment by ayman-m** [comment]
@ChanochShayner, I am not sure what's wrong with Test/conf.json, and why it's causing a merge conflict, one more issue, I am currently  using a non-demisto docker image for the MinIO integration (minio:latest), shall I submit another PR for a docker build file ?

**Review by ChanochShayner — COMMENTED** [review]
You need to be consistent with the file versions - currently, all files are from version 6, except:
`Packs / DevSecOps / Integrations / GitLab / GitLab.yml`
`Packs / DevSecOps / Layouts / layout-DevSecOps_New_Git_Issue.json` - (I think this incident type no longer exists, if so please remove this layout.)
`Packs / DevSecOps / Layouts / layout-DevSecOps_New_Git_PR.json`
If you want to change the layout to version 6 you can use the command that exists in sdk (I explained it in the previous comment)

**Comment by ayman-m** [comment]
> You need to be consistent with the file versions - currently, all files are from version 6, except:
> `Packs / DevSecOps / Integrations / GitLab / GitLab.yml`
> `Packs / DevSecOps / Layouts / layout-DevSecOps_New_Git_Issue.json` - (I think this incident type no longer exists, if so please remove this layout.)
> `Packs / DevSecOps / Layouts / layout-DevSecOps_New_Git_PR.json`
> If you want to change the layout to version 6 you can use the command that exists in sdk (I explained it in the previous comment)

**Comment by ayman-m** [comment]
MinIO integration requires the "minio" package , I am referencing a local (non xsoar) docker image in the code

**Comment by ChanochShayner** [comment]
Unassigning myself, @guykeller @yaakovi FYI.

**Review by yaakovi — COMMENTED** [review]
PR is pending for https://github.com/demisto/content/pull/12148

**Comment by ayman-m** [comment]
> PR is pending for #12148

@yaakovi FYI #12148 is now merged

**Review by yaakovi — APPROVED** [review]

**Pull request merged**